spec/formatting.md

```diff
@@ -505,14 +505,19 @@ _Formatting_ is a mostly implementation-defined process,
 as it depends on the implementation's shape for resolved values
 and the result type of the formatting.
 
-Formatting errors MAY be emitted during _formatting_,
-as formatting is not necessarily defined on every resolved value.
-A formatter MAY provide a value to use in such a case instead of a _fallback value_.
-
-_Formatting_ MAY produce formatted messages with the following data types,
-as well as any others:
-
-- A single concatenated string.
+Resolved values cannot always be formatted by a given implementation.
+When such an error occurs during _formatting_, 
+an implementation SHOULD emit a _formatting error_ and produce a 
+_formatting fallback value_ for the _placeholder_ that produced the error.
+An implementation MAY substitute a value to use instead of a _falback value_.
```

Comment:
```suggestion
_fallback value_ for the _placeholder_ that produced the error.
A formatting function MAY substitute a value to use instead of a _fallback value_.
```

I'm pretty sure that we don't need to specify _formatting fallback value_, but just _fallback value_.

Also, the intent of the original text was to allow for a formatting function to supply a substitution. By default we avoid revealing any runtime values as doing so could have security concerns, but a formatter that's confident about its input could opt to use a different fallback.

For instance, a dynamic message reference formatter could use the resolved message identifier as its fallback value rather than the name of the variable. This would allow a missing message to format as something like `{action.cancel}` rather than `{$msgId}`.

Reply:
The difference here is something else, though. The original text said "an implementation", which means the message formatting engine. My text says "a formatting function", which (a) allows the function to provide an alternate fallback value (just as your comment spells out) and (b) means that implementations have to provide the ability for a formatting function to supply the fallback value. 

In fact, that seems like a potential requirement or a potential option, so I'll clarify the text some more.......

Reply:
The original text actually said "formatter"; your PR currently says "implemantation"; it's only my suggestion above that uses "formatting function".

Reply:
My bad. I'm reading this comment wrong.